File: Runtime/src/Core/CoreScript/BCL/IteratorBlockEnumerable.js
///////////////////////////////////////////////////////////////////////////////
// IteratorBlockEnumerable

ss.IteratorBlockEnumerable = function#? DEBUG IteratorBlockEnumerable$##(getEnumerator, $this) {
    this._getEnumerator = getEnumerator;
    this._this = $this;
};

ss.IteratorBlockEnumerable.prototype = {
    getEnumerator: function#? DEBUG IteratorBlockEnumerable$getEnumerator##() {
        return this._getEnumerator.call(this._this);
    }
};

ss.IteratorBlockEnumerable.registerClass('ss.IteratorBlockEnumerable', null, ss.IEnumerable);

File: Runtime/CoreLib/Collections/Generic/IteratorBlockEnumerable.cs
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;
using System.Text;

namespace System.Collections.Generic
{
	[Imported(ObeysTypeSystem = true)]
	[IncludeGenericArguments(false)]
	[ScriptNamespace("ss")]
	internal class IteratorBlockEnumerable<T> : IEnumerable<T> {
		public IEnumerator<T> GetEnumerator() { return null; }
		IEnumerator IEnumerable.GetEnumerator() { return null; }
	}
}
